src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -527,4 +556,29 @@ public SCMFileSystem build(@NonNull SCMSource source, @NonNull SCMHead head,
             return build(owner, source.build(head, rev), rev);
         }
     }
+
+    public abstract static class Builder2 extends Builder {
```

Comment:
Oki, well I added a new overridable method, instead of a new class with abstract method. Although I couldn't find a suitable place where to use the Util.isOverridden,  if the target library now doesnt implement the new method, it will be automatically called the base method that will delegate to old method. Is that OK ?

Reply:
OK if all calls to `build` go through `SCMFileSystem.of`, which may be true. Still make the original method not be `abstract` (have it throw `AbstractMethodError`) so an impl need only override the new overload.

Reply:
Ok I removed the abstract modifier and added a throw with notice to implement the new version. Also changed the other call to build that should be calling the new implementation.  I will try to figure out how to make the draft PR in the workflow-csp and git plugin once i figure out how this incremental feature works